eclair-core/src/test/scala/fr/acinq/eclair/TestUtils.scala

```diff
@@ -104,7 +106,13 @@ object TestUtils {
       eventStream.publish(DummyEvent())
       assert(listener.msgAvailable)
     }
+  }
 
+  def waitFor(duration: FiniteDuration): Unit = {
+    val now = TimestampMilli.now()
+    eventually(PatienceConfiguration.Timeout(duration * 2), PatienceConfiguration.Interval(50 millis)) {
+      assert(TimestampMilli.now() - now > duration)
+    }
```

Comment:
No it shouldn't, this should be using a spin lock (or something equivalent) and should release the thread after each test (so that other tests can run on that thread between checks), whereas `Thread.sleep` blocks the thread for the whole duration for everyone.

Reply:
According to the doc of `eventually`, it attempts the check then sleeps (I assume with `Thread.sleep`) for the interval duration, then rinse & repeat. So it's not as good as I hoped, but it's still better than `Thread.sleep`, because instead of blocking the thread for the whole duration, we block it for smaller chunks, which gives the opportunity for the process scheduler to run something else on this thread regularly.

Reply:
Yes, but I think this saves thread time for the _additional_ thread that executes whatever code is in the `eventually`. The _caller_ thread still has to wait. Maybe I'm completely wrong.